proxy/hdrs/HTTP.cc

```diff
@@ -1163,6 +1169,37 @@ http_parser_parse_req(HTTPParser *parser, HdrHeap *heap, HTTPHdrImpl *hh, const
   return ret;
 }
 
+ParseResult
+validate_hdr_request_target(int method_wk_idx, URLImpl *url)
+{
+  ParseResult ret = PARSE_RESULT_DONE;
+  int host_len;
+  url->get_host(&host_len);
+  int path_len;
+  const char *path = url->get_path(&path_len);
+  int scheme_len;
+  url->get_scheme(&scheme_len);
+
+  if (host_len == 0) {
+    if (path_len == 1 && path[0] == '*') { // asterisk-form
+      // Skip this check for now because URLImpl can't distinguish '*' and '/*'
+      // if (method_wk_idx != HTTP_WKSIDX_OPTIONS) {
+      //   ret = PARSE_RESULT_ERROR;
+      // }
+    } else { // origin-form
+      // Nothing to check here
+    }
+  } else if (scheme_len == 0 && host_len != 0) { // authority-form
+    if (method_wk_idx != HTTP_WKSIDX_CONNECT) {
+      ret = PARSE_RESULT_ERROR;
+    }
+  } else { // absolute-form
+    // Nothing to check here
```

Comment:
maybe it's redundant but could we double check here that `path[0] == '/'` as it's the very defining characteristic of the absolute-form per RFC 9112?

Reply:
oh, understood!